router/router.go

```diff
@@ -71,7 +71,12 @@ func New(log logrus.FieldLogger, options ...Option) Router {
 	}
 
 	if r.healthEndpoint != "" {
-		r.Use(HealthCheck(r.healthEndpoint, r.healthHandler))
+		h := r.healthHandler
```

Comment:
I think that there is an issue with adding a route and then adding middlewares. Which also means that even if we move the tracing one up (which would suck) ~ it stops users of the lib from adding more middlewares.

Reply:
You're completely right: 
>panic: chi:  all middlewares must be defined before routes on a mux

I'm gonna drop this PR and fix the bug on the Middleware then!
Thanks for the catch!